pymongo/asynchronous/pool.py

```diff
@@ -1395,6 +1412,20 @@ async def checkin(self, conn: AsyncConnection) -> None:
                     # Pool.reset().
                     if self.stale_generation(conn.generation, conn.service_id):
                         close_conn = True
+                    # If in backoff state, check the conn's readiness.
+                    elif self._backoff:
+                        # Set a 1ms read deadline and attempt to read 1 byte from the connection.
+                        # Expect it to block for 1ms then return a deadline exceeded error. If it
+                        # returns any other error, the connection is not usable, so return false.
+                        # If it doesn't return an error and actually reads data, the connection is
+                        # also not usable, so return false.
+                        conn.conn.get_conn.settimeout(0.001)
+                        close_conn = True
+                        try:
+                            conn.conn.get_conn.read(1)
+                        except Exception as _:
+                            # TODO: verify the exception
+                            close_conn = False
```

Comment:
2 comments: 
1) I believe this logic needs to move to connection checkout. Here in connection check in we already know the connection is useable because we're checking in back in after a successful command.
2) Instead of a 1ms can we reuse the existing _perished() + conn_closed() methods?

Reply:
Done